feature-libs/cart/import-export/core/services/import-csv.service.ts

```diff
@@ -49,12 +36,59 @@ export class ImportCsvService {
   }
 
   loadCsvData(file: File): Observable<string[][]> {
-    return this.loadFile(file).pipe(
-      map((res) => this.readCsvData(res as string))
-    );
+    return this.importService
+      .loadTextFile(file)
+      .pipe(map((res) => this.readCsvData(res as string)));
+  }
+
+  isReadableFile(
+    isDataParsable?: (data: string[][]) => boolean
+  ): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> => {
+      const errors: ValidationErrors = {};
+      const file: File = control.value[0];
```

Comment:
This service makes a hard assumption on the value type inside the value control.
I'm in favor of:
- pass only a `File` to this service, so it has no notion of the form structure
- in the component which creates the form: create the actual validator and reuse there this service method inside
- you might want to rename this method eg. to `validateData(file: File): Observable<...>`
- it would be nice to narrow down the return type of this method (to be precise which errors it might recognize), for example Observable<CsvFileValidationErrors | null>
i.e. 
```
interface CsvFileValidationErrors {
  empty: boolean,
  tooManyEntries?: { .... }
  ....
}
```

Reply:
I'd like to do it most update-friendly for an upgrade to have possibility usage of multi-file if someone wants that